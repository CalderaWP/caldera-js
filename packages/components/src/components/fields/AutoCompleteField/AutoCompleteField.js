import {FormTokenField} from '@wordpress/components';
import { BaseControl } from '@wordpress/components';
import React from "react";
import {fieldClassNames} from "../util";
import {SelectField} from '../SelectField/SelectField';
export const AutoCompleteField = (props) => {
	const {
		label,
		description,
		fieldId,
		placeholder,
		required,
		value,
		onChange,
		onBlur,
		options
	} = props;
	return (
		<BaseControl
			id={fieldId}
			label={label}
			help={description}
		>
			<FormTokenField
				className={fieldClassNames('autocomplete')}
				id={fieldId}
				value={ value }
				suggestions={ options }
				onChange={ tokens => onChange(tokens) }
				placeholder={placeholder}
			/>
		</BaseControl>
	);
}

AutoCompleteField.propTypes = SelectField.propTypes;
AutoCompleteField.defaultProps = SelectField.defaultProps;
