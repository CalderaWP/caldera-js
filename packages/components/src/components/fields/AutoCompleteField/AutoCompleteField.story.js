import React from 'react';
import { storiesOf } from '@storybook/react';
import { AutoCompleteField } from './AutoCompleteField';

const onChange = newValue => {
	console.log(newValue);
};
const onBlur = event => {};

const STORY_NAME_AUTOCOMPLETE = 'AutoCompleteField';
storiesOf(STORY_NAME_AUTOCOMPLETE, module).add(
	'With no value selected and  placeholder',
	() => (
		<AutoCompleteField
			label={'Select A Hat'}
			onChange={onChange}
			description={'selection of hats'}
			fieldId={'selection-hats'}
			required={true}
			options={['a', 'b', 'c']}
			value={[]}
		/>
	)
);
storiesOf(STORY_NAME_AUTOCOMPLETE, module).add(
	'With no value selected ',
	() => (
		<AutoCompleteField
			label={'Select A Hat'}
			onChange={onChange}
			description={'selection of hats'}
			fieldId={'selection-hats'}
			required={true}
			options={['a', 'b', 'c']}
			value={['b']}
		/>
	)
);
storiesOf(STORY_NAME_AUTOCOMPLETE, module).add('With two values selects', () => (
	<AutoCompleteField
		label={'Select A Hat'}
		onChange={(change) => console.log(change)}
		description={'selection of hats'}
		fieldId={'selection-hats'}
		required={true}
		value={['One', 'Three']}
		options={['One', 'Two', 'Three']}
	/>
));
