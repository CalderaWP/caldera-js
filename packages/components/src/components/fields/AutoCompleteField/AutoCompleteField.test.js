import * as React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { AutoCompleteField } from './AutoCompleteField';
import {SelectField} from "../../..";

describe('AutoCompleteField', () => {
	let onChange;
	beforeEach(() => {
		onChange = jest.fn();
	});
	it('should match snapshot', () => {
		const component = renderer.create(
			<AutoCompleteField
				label={'Select A Hat'}
				onChange={onChange}
				description={'selection of hats'}
				fieldId={'auto-complete-hats'}
				required={true}
				multiple={true}
				options={['a', 'Boom', 'stick']}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();

	});

	it.skip('Fires on change event', () => {
		const component = mount(
			<AutoCompleteField
				label={'Select A Hat'}
				onChange={(values) => {
					onChange(values);
				}}
				description={'selection of hats'}
				fieldId={'selection-hats'}
				required={true}
				multiple={true}
				options={['a', 'Boom', 'stick']}
			/>
		);

		const event = { target: { value: 'Two' } };
		expect(component.find('input').length).toBe(1);
		component.find('input').simulate('change', event);
		expect(onChange.mock.calls.length).toBe(1);
		expect(onChange.mock.calls[0][0]).toBe('Two');
	});

});
