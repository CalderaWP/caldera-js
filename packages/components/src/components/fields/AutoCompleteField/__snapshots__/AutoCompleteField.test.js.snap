// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoCompleteField should match snapshot 1`] = `
<div
  className="components-base-control"
>
  <div
    className="components-base-control__field"
  >
    <label
      className="components-base-control__label"
      htmlFor="auto-complete-hats"
    >
      Select A Hat
    </label>
    <div
      className="components-form-token-field"
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      tabIndex="-1"
    >
      <label
        className="components-form-token-field__label"
        htmlFor="components-form-token-input-0"
      >
        Add item
      </label>
      <div
        className="caldera-field caldera-field-autocomplete components-form-token-field__input-container"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
      >
        <input
          aria-activedescendant={undefined}
          aria-autocomplete="list"
          aria-describedby="components-form-token-suggestions-howto-0"
          aria-expanded={false}
          aria-owns={undefined}
          autoCapitalize={undefined}
          autoComplete={undefined}
          className="components-form-token-field__input"
          disabled={false}
          id="components-form-token-input-0"
          onBlur={[Function]}
          onChange={[Function]}
          role="combobox"
          size={1}
          type="text"
          value=""
        />
      </div>
      <div
        className="screen-reader-text"
        id="components-form-token-suggestions-howto-0"
      >
        Separate with commas
      </div>
    </div>
  </div>
  <p
    className="components-base-control__help"
    id="auto-complete-hats__help"
  >
    selection of hats
  </p>
</div>
`;
